plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -312,12 +317,38 @@ public static String quotedTableName(SchemaTableName name)
 
     private static String quotedName(String name)
     {
-        if (SIMPLE_NAME.matcher(name).matches()) {
+        if (UNQUOTED_IDENTIFIER_PATTERN.matcher(name).matches()) {
```

Comment:
That changes semantics of the method.

previously, for `My_Table` we would output `"My_Table"`.
now we output `My_Table` without quotes.

if the table name is actually My_Table, it needs to be referenced as `"My_Table"` in SQL,
so the output of this command no longer can be pasted into SQL.

Please revert the change here

Reply:
You indeed point out a bug, that also applies to the `fromColumnToIdentifier` method. It is however the same semantics: we are taking something from metadata (a column or table name), and need to ensure that it can be pasted in an SQL editor, respecting SQL identifier semantics.